### DISCUSSION TITLE
Deep learning in sktime, sktime-dl integration

### DISCUSSION THREAD

**Discussion by ilkersigirci** [alan-turing-institute/sktime#2046] [Q&A] [answered]
Will there be any specified date for merging with sktime-dl? I couldn't see any information in the roadmap of both library?

**Comment by fkiraly** [comment]
This is a major topic in current dev discussions.

0.10.0 was mainly about a major refactor of CI/CD and test framework, which has now unlocked working on the integration of "functionality formerly in `sktime-dl`, packages that follow `sktime` interfaces, and plugins for `sktime` in general.

There is not a specified date since we are still in design/architecture phase, but there will be exploratory PR very soon, then to be followed by more systematic implementation activities.

If interested to be involved: join the dev community on slack or in the Friday meetings, and/or ping @chrisholder, @MatthewMiddlehurst, @jasonlines, @TonyBagnall who are driving this.

**Comment by TonyBagnall** [comment, marked as answer]
hi, the approximate road map is this
1. PR to include tslearn as a soft dependency (which is dependent on tensorflow). @chrisholder putting it up today hopefully, should go in if it does not break our CI/CD
2. PR to make tensorflow a soft dependency, with toy usage, to make sure it all works.
3. PR to port over first classifier, needs a bit of refactoring, I should be able to do this in march but there are a few design issues to resolve
4. Incremental PR over, should be fairly fast after (3)
realistically, should get it all in over the summer, but it might be quicker, depends on time.

**Reply by ilkersigirci** [reply]
Thank you for the quick and detailed response. I have recently started to use sktime. In the near future, I would to help the development process of this great library.

**Reply by TonyBagnall** [reply]
great, we literally just had a meeting on the topic, hence the speedy reply. All help greatly appreciated!
